src/appleseed/renderer/modeling/object/rectobject.h

```diff
@@ -0,0 +1,126 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Esteban Tovagliari, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.renderer headers.
+#include "renderer/modeling/object/iobjectfactory.h"
+#include "renderer/modeling/object/proceduralobject.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/types.h"
+
+// appleseed.main headers.
+#include "main/dllsymbol.h"
+
+// Forward declarations.
+namespace foundation    { class IAbortSwitch; }
+namespace renderer      { class ParamArray; }
+namespace renderer      { class ShadingRay; }
+
+namespace renderer
+{
+
+//
+// A rect object.
+//
+// The rect is assumed to be centered at the origin.
```

Comment:
self-review: this comment is not very useful...

Reply:
Here too, it's not assumed, it _is_. I would suggest not only to keep this comment (because it is useful, it's not obvious that the rectangle is centered at the origin, one of its corner could be at the origin instead) but also to augment it with information about the plane in which the rectangle lies (is it X-Y? X-Z?).